### DISCUSSION TITLE
Deleting values from a repeater?

### DISCUSSION THREAD

**Discussion by sriver** [writer/writer-framework#168] [General]
I'm trying to implement a repeater structure for a dict where each row consists of a key and a button for deleting that entry. However, it seems that the StateProxy-based structure cannot be (easily) manipulated through an associated event handler. Any ideas how this could be done?

**Comment by FabienArcellier** [comment]
Hi,

Last time I have to implement remove action on a repeater, I have used a dict comprehension to rebuild the dictionnary.
It has to effect to send a new dictionnary to the frontend.

```python
def serial_number_remove(state, context):
    items = state["items"].state
    items = {k: v for k, v in items.items() if k != context["item"]}
    state["items"] = items
```

Regards,

**Reply by sriver** [reply]
Thanks. Interestingly it seems that this straight dict comprehension doesn't work in my case, where the state structure is actually a dict of dicts, ie. each value v is a dict. I got it working by re-constructing each value dict in dict comprehension: 
    my_items = {k: {'title': v['title'], 'href': v['href']} for k, v in my_items.items() if k != context["itemId"]}

**Reply by FabienArcellier** [reply]
have you try using `copy.copy` or `copy.deep_copy` ? it may be easier to maintain

**Reply by sriver** [reply]
Using copy would have been much more elegant, but unfortunately it doesn't seem to work either. Maybe there's something around the underlying mutation that does not get triggered properly? Anyway, I'm good with the reconstruction solution above.
